pkg/kubelet/kubelet_pods.go

```diff
@@ -1644,6 +1644,24 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 				FinishedAt:  metav1.NewTime(cs.FinishedAt),
 				ContainerID: cid,
 			}
+
+		case cs.State == kubecontainer.ContainerStateUnknown &&
```

Comment:
@deads2k appreciate your help chasing these down.

I think we are reaching point of 'unit test would be really helpful', can you look at `TestGenerateAPIPodStatusWithReasonCache` and see if you can extend to cover this scenario?  I think it will hit your flow changes.